### ISSUE TITLE
bug: events.PutEvents fails with custom bus name using boto3

### ISSUE THREAD

**Issue by sjbitcode** [localstack/localstack#8261]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Current Behavior

The LocalStack Docker image >=2.0.0 throws a 500 `InternalError` when calling the `events.PutEvents` operation via boto3 **with a custom event bus name** (i.e. the event bus name is not `default`).

(docker-compose logs)
```
ERROR --- [   asgi_gw_0] l.aws.handlers.logging     : exception during call chain: 'customBusName'
 INFO --- [   asgi_gw_0] localstack.request.aws     : AWS events.PutEvents => 500 (InternalError)
```

(boto3 error traceback)
```
Traceback (most recent call last):
  File "eb_publish.py", line 6, in <module>
    response = eb.put_events(
  File "/Users/sangeeta.jadoonanan/.pyenv/versions/hydra/lib/python3.8/site-packages/botocore/client.py", line 530, in _api_call
    return self._make_api_call(operation_name, kwargs)
  File "/Users/sangeeta.jadoonanan/.pyenv/versions/hydra/lib/python3.8/site-packages/botocore/client.py", line 960, in _make_api_call
    raise error_class(parsed_response, operation_name)
botocore.exceptions.ClientError: An error occurred (InternalError) when calling the PutEvents operation (reached max retries: 4): exception while calling events.PutEvents: 'customBusName'
```

### Expected Behavior

I expected the `events.PutEvents` operation to get called successfully and result in a 200 status code with a custom event bus name, like I've tested with **LocalStack 1.4**.

(docker-compose logs running LocalStack 1.4)
```
INFO --- [   asgi_gw_0] localstack.request.aws     : AWS events.PutEvents => 200
```

### How are you starting LocalStack?

With a docker-compose file

#### How are you starting localstack (e.g., `bin/localstack` command, arguments, or `docker-compose.yml`)

Here's my `docker-compose.yml`:
```yaml
version: "3.4"

services:
  eventbridge:
    image: localstack/localstack
    container_name: hydra_eventbridge
    ports:
      - 4566:4566
    environment:
      - EAGER_SERVICE_LOADING=1
      - SERVICES=events
```

To start LocalStack, I run `docker-compose up`.

The [image versions `2.0.0`](https://hub.docker.com/layers/localstack/localstack/2.0.0/images/sha256-2d0861a7fd281bb4f8a8404d8249ab4aed278c5ac8bdc55f8c246399e4bffcb8?context=explore) and higher fail when passing a custom bus name. The highest working [image version is `1.4`](https://hub.docker.com/layers/localstack/localstack/1.4/images/sha256-4a966e42eff4bbeec44afe13c2d24a67954742b191caae7c15e56186cc0b9ed8?context=explore).

You can try toggling the versions in the `docker-compose.yml` to verify:
```
image: localstack/localstack:1.4
```

and

```
image: localstack/localstack:2.0.0
```

#### Client commands (e.g., AWS SDK code snippet, or sequence of "awslocal" commands)

Here's my Python boto3 snippet to publish an event with a custom bus name:
```python
import boto3

eb = boto3.client(service_name="events", endpoint_url="http://localhost:4566")

eb.put_events(
    Entries=[{
        "Source": "test-source",
        "Detail": "test-detail",
        "DetailType": "test-detail-type",
        "EventBusName": "customBusName"
    }]
)
```

### Environment

```markdown
- OS: macOS Ventura 13.2
- LocalStack: 2.0.0
- Python: 3.8.13
- boto3==1.26.127
- botocore==1.29.127
```

### Anything else?

**TL;DR:** The `1.4` localstack Docker image works for me, and anything from `2.0.0` and higher fails.

I noticed this issue with the latest `localstack` Docker image and worked backwards until I found that the breaking image version was `2.0.0`.

Also worth noting that all versions work when I use the default bus name of `default`.

**Comment by bentsku** [comment]
Hi @sjbitcode and thanks for your report!

There's a PR linked to fix the behaviour and allow publishing to nonexistent custom EventBus. AWS also allows publishing to a custom EventBus that does not exist, even though it won't do anything with it.

In the meantime, not knowing the context of your usage of LocalStack, maybe you can create the EventBus before publishing to it? That could resolve the issue.

**Comment by sjbitcode** [comment]
Thanks @bentsku for the speedy PR and also for clarifying that the issue was really the nonexistent bus!

I suppose I could create the bus via boto3 or aws cli if I'm running LocalStack. My actual code assumes the bus already exists, so LocalStack came in handy to emulate that easily. I can also pin the LocalStack image to 1.4 until this change gets released.

**Comment by lesandiz** [comment]
It looks like there is a regression on this. When using latest images (2.3.1 at the time of writing) and trying to put events into an existing custom bus, it fails with error `AWS events.PutEvents => 500 (InternalError)`

## To reproduce

```
aws events create-event-bus \
    --name demo \
    --endpoint-url http://localhost:4566

aws events put-events \
    --entries file://demo.json \
    --endpoint-url=http://localhost:4566
```

And `demo.json` content

```
[
    {
      "Source": "com.mycompany.myapp",
      "EventBusName": "demo",
      "Detail": "{ \"key1\": \"value1\", \"key2\": \"value2\" }",
      "Resources": [
        "resource1",
        "resource2"
      ],
      "DetailType": "myDetailType"
    }
  ]
```
